go/connector/run.go

```diff
@@ -225,38 +225,40 @@ func runCommand(
 		},
 	}
 
+	var group = sync.WaitGroup{}
 	var conn net.Conn
 	// If port is specified, use TCP socket
 	if port != "" {
+		// We're going to need to await 3 separate async tasks:
+		// The attempt to dial, and the two copy operations in
+		// and out of the established connection.
+		group.Add(3)
 		// Routine dialing a connection to connector
 		go func() {
 			// Try to connect to the connector with a retry mechanism
-			var connectDeadline = time.Now().Add(time.Second * 10)
-			var err error
-			for {
-				conn, err = net.Dial("tcp", localAddress)
-				if err == nil {
-					break
-				}
-
-				if time.Now().After(connectDeadline) {
-					fe.onError(fmt.Errorf("dialing connection to %s: %w", localAddress, err))
-					return
-				} else {
-					// Retry after a short wait
-					time.Sleep(1 * time.Second)
-				}
+			// Don't retry on context cancellation
+			// Retry after a short wait
+			conn, connErr := connectTCP(ctx, conn, localAddress)
+			group.Add(-1)
```

Comment:
nit: You should probably use `group.Done()` for this, although it's literally a helper alias for `Add(-1)` so this works fine too.